src/coreclr/Directory.Build.targets

```diff
@@ -2,36 +2,28 @@
   <Import Project="..\..\Directory.Build.targets" />
 
   <ItemGroup>
-    <BuiltBinary Include="$(TargetPath)" />
-    <_AllPdbs Include="@(BuiltBinary -> '%(RootDir)%(Directory)%(Filename).pdb')">
-      <FolderName>$([System.IO.Directory]::GetParent(`%(Identity)`).Name)</FolderName>
-    </_AllPdbs>
+    <_DebugSymbolToMove Include="@(DebugSymbolsProjectOutputGroupOutput->Metadata('FinalOutputPath'))"
+                        FolderName="$([System.IO.Directory]::GetParent('%(Identity)').Name)" />
+    <_DebugSymbolToMoveToExclude Include="@(_DebugSymbolToMove->WithMetadataValue('FolderName', 'aotsdk'))" />
+    <_DebugSymbolToMoveToUpdate Include="@(_DebugSymbolToMove)"
+                                Exclude="@(_DebugSymbolToMoveToExclude)" />
+    <!-- Remove the FolderName metadata for projects that don't reside under an aotsdk folder. -->
+    <_DebugSymbolToMove Update="@(_DebugSymbolToMoveToUpdate)"
+                        FolderName="" />
+    <_DebugSymbolToMove Update="@(_DebugSymbolToMove)"
+                        Destination="$(RuntimeBinDir)PDB\%(FolderName)\%(Filename)%(Extension)" />
   </ItemGroup>
 
-  <!-- Target used to consolidate all PDBs into a single location, except the aotsdk
-       PDBs, as S.P.C will overwrite the coreclr PDBs -->
+  <!-- Consolidate all PDBs into a single location, except the aotsdk PDBs, as S.P.C will overwrite the coreclr PDBs. -->
   <Target Name="MoveSymbolFiles"
-          AfterTargets="Build"
-          Condition="Exists(@(_AllPdbs))"
-          Inputs="@(_AllPdbs)"
-          Outputs="@(_AllPdbs -> '$(RuntimeBinDir)PDB/%(Filename).pdb')">
-
-    <Move SourceFiles="@(_AllPdbs)"
-          DestinationFolder="$(RuntimeBinDir)PDB"
-          Condition="'%(_AllPdbs.FolderName)' != 'aotsdk'" />
-
-  </Target>
-
-  <Target Name="MoveAotSymbolFiles"
-          AfterTargets="Build"
-          Condition="Exists(@(_AllPdbs))"
-          Inputs="@(_AllPdbs)"
-          Outputs="@(_AllPdbs -> '$(RuntimeBinDir)PDB/aotsdk/%(Filename).pdb')">
-
-    <Move SourceFiles="@(_AllPdbs)"
-          DestinationFolder="$(RuntimeBinDir)PDB/aotsdk"
-          Condition="'%(_AllPdbs.FolderName)' == 'aotsdk'" />
-
+          AfterTargets="CopyFilesToOutputDirectory"
+          DependsOnTargets="_CheckForCompileOutputs"
```

Comment:
Is there any non-internal target that we can hang off of? Like this target is implied by `CopyFilesToOutputDirectory`. Is the AfterTargets here not enough?